### ISSUE TITLE
A Rigorous Benchmark of Quantum Kernel Classifiers for Breast Cancer Diagnosis

### ISSUE THREAD

**Issue by JeromeB7** [qiskit-advocate/qamp-2025#12]
# QAMP Project Proposal: A Rigorous Benchmark of Quantum Kernel Classifiers for Breast Cancer Diagnosis

## Description
This project provides a rigorous and practical benchmark of a Quantum Support Vector Classifier (QSVC) for breast cancer diagnosis. Using the well-known Wisconsin Breast Cancer dataset from Scikit-learn, this project goes beyond a simple proof-of-concept by implementing a robust validation framework. While this dataset is a common benchmark, the rigorous noise-aware validation of quantum kernels is still underexplored and offers a valuable contribution to QML education. The core methodology involves applying Principal Component Analysis (PCA) for feature reduction, followed by a 10-fold cross-validation to compare the performance of the Qiskit-based QSVC against a classical Support Vector Machine (SVM).

Crucially, the project will also analyze the impact of realistic hardware noise on the quantum model's performance by running the entire validation pipeline on a simulated noisy backend based on a real IBM Quantum device. The goal is to create a high-quality educational resource and case study that demonstrates professional best practices for validating a QML model and assessing its viability in the NISQ era. Although this is a well-defined dataset, the rigorous framework developed can serve as a blueprint for future, more complex biomedical QML studies.

## Deliverables
**Primary Deliverables:**
- [ ] A comprehensive, open-source Jupyter Notebook implementing the full pipeline: PCA, 10-fold cross-validation for both classical SVM and the Qiskit QSVC.
- [ ] A detailed analysis section within the notebook comparing the performance (accuracy, precision, recall) of the ideal (noiseless) QSVC, the noisy QSVC, and the classical SVM baseline.
- [ ] A tutorial-style blog post (for platforms like Medium or the Qiskit Blog) explaining the methodology, the importance of rigorous validation, and the impact of noise on the results.

**Minimal Viable Product (MVP):**
- [ ] A single, well-documented Jupyter notebook that performs a simple train/test split and successfully compares the accuracy of a classical SVM and an ideal (noiseless) QSVC on the prepared dataset.

## Mentors
**Type of mentor sought:** I am seeking a mentor with experience in Qiskit Machine Learning and a strong understanding of standard ML validation techniques (like cross-validation and benchmarking). Experience with Qiskit's noise models or practical knowledge of quantum hardware noise would be highly beneficial for the advanced goals of this project. Guidance on open-source best practices would also be appreciated.

**Desired mentoring style:** I would appreciate regular check-ins (ideally weekly) to discuss progress and challenges. I am looking for a mentor who can provide actionable feedback on both the code and the scientific methodology, helping to ensure the project is robust and well-documented.

## Mentee
**Name:** Jerome B
**GitHub:** @JeromeB7
**What I do:** I am final year computer science undergrad student, passionate about the Biomedical applications of Quantum Computing. I am proficient in Machine Learning and moving forward towards Quantum Machine Learning. I have previously also worked on developing a Hybrid Quantum Model for Protein Fold Prediction.

**Learning goals:** I hope to deepen my understanding of Quantum Machine Learning, learn best practices for robust model validation like K-Fold cross-validation, and gain practical insight into the impact of hardware noise on the performance of QML algorithms.

**Additional Information:**
- **Time commitment:**  6-8 hours per week
- **Prerequisites:** Python, Qiskit, Pandas, Scikit-learn, and a foundational understanding of quantum circuits.
- **Resources:** IBM Quantum backends (provided via Advocate access), Scikit-learn libraries and datasets.
---

Checklist after creating this issue:
- [x] Submit the 2025 QAMP Application Form if I am a mentee (link in Slack and Discord)

**Comment by LAKHSHYAA** [comment]
hi @JeromeB7 I would love to work on this project with you as a mentee, if possible.

**Comment by sgayathridevi** [comment]
Hi JeromeB7 , I would like to work on this project as a mentee.

**Comment by JeromeB7** [comment]
> Hi [JeromeB7](https://github.com/JeromeB7) I am excited to be Mentor for this project

Hi @akortheanchor, thank you! I'm very excited about this project and look forward to working under your guidance.

**Comment by chanakyaVBSL** [comment]
@JeromeB7 I would like to join as a mentee, I have experience in Quantum Machine Learning by using frameworks like qisikit-machine-learning, pennylane and pytorch.

**Comment by akortheanchor** [comment]
Hi @JeromeB7  : There are only two available mentees : [DanielCondeTorres](https://github.com/DanielCondeTorres) and [Bhuvana-666](https://github.com/Bhuvana-666) and would like to take [friedsam](https://github.com/friedsam) 
As a mentor i would like to suggest the above three Mentees to work with ........

**Comment by DanielCondeTorres** [comment]
Hi @Bhuvana-666, @JeromeB7, @friedsam and the mentor @akortheanchor, excited to work with you!

**Comment by JeromeB7** [comment]
Hi @Bhuvana-666, @DanielCondeTorres and @akortheanchor our mentor, I am also really excited to work with you all.

**Comment by DanielCondeTorres** [comment]
> Hello, in order to communicate and work together, do you think it would be a good idea to create a Slack channel and a GitHub repository where we can collaborate? Best regards

Sorry, Slack is not necessary, there is a discord channel for the team!

**Comment by HichanBen** [comment]
Hi @Bhuvana-666, @DanielCondeTorres and @akortheanchor our mentor, I am also really excited to work with you all.

**Comment by HichanBen** [comment]
Hi @JeromeB7 , I would love to work on this project with you as a mentee, if possible.

**Comment by JeromeB7** [comment]
@akortheanchor Sir, Kindly join our discord group channel by clicking #12 at qamp-groups so that we can have our communications over there.

**Comment by akortheanchor** [comment]
> [@akortheanchor](https://github.com/akortheanchor) Sir, Kindly join our discord group channel by clicking [#12](https://github.com/qiskit-advocate/qamp-2025/issues/12) at qamp-groups so that we can have our communications over there.

I have already accepted the invite link, but cant see the group 12 in the server

**Comment by Bhuvana-666** [comment]
# Checkpoint 2
### [Checkpoint 2 Report (#12).pdf](https://github.com/user-attachments/files/24243088/Checkpoint.2.Report.12.pdf)
# A Rigorous Benchmark of Quantum Kernel Classifiers for Breast Cancer Diagnosis 
---

## Project overview

This project aims to benchmark the performance of Quantum Machine Learning (QML) against classical state-of-the-art methods for the binary classification of breast cancer (Malignant vs Benign). The primary research question is whether mapping classical data into a high-dimensional quantum Hilbert space using Quantum Kernels offers a tangible decision boundary advantage over classical Support Vector Machines (SVMs) in the NISQ (Noisy Intermediate-Scale Quantum) era.

The complete source code, including the data pipeline, custom feature maps, and experiment logs, is available in our GitHub repository:[https://github.com/JeromeB7/qamp-2025-qml-cancer-benchmark]( https://github.com/JeromeB7/qamp-2025-qml-cancer-benchmark)


---

## Progress made

We implemented and evaluated Classical and Quantum models for Breast Cancer Classification on a single dataset using noiseless quantum simulations.
- Established **Linear, RBF and Polynomial SVM** baselines for comparison.
- Implemented QSVMs with **Fourier and QAOA**-inspired feature maps, producing stable and structured decision boundaries.
- Evaluated trainable kernels (**data re-uploading and HEA**), highlighting the advantages of hardware-efficient designs.
- Explored **Haar-random kernels**, observing reduced generalization with increased randomness.
- Began preliminary work toward **QFT-based architectures.**
---

## Explorations & Experiments

Using the WDBC dataset, we applied PCA to reduce the 30 features to 7 components while retaining ~90% variance, enabling shallow quantum circuits. Within this setup, we explored classical and quantum models by varying feature encoding, kernel structure, randomness, and trainability including SVMs, QSVMs (Fourier and QAOA), trainable circuits (data re-uploading and HEA), Haar-random kernels, and preliminary QFT-based approaches, using noiseless simulations to isolate architectural effects.

---

## Key Insights & Learnings

Our benchmarking framework compares classical baselines against a custom Quantum Support Vector Machine (QSVM) architecture.

**A. Classical Baselines** 
We deployed Linear SVMs (to test for linear separability) and RBF (Radial Basis Function) SVMs (the classical "gold standard" for non-linear classification). 

These serve as the control group to validate the dataset's complexity and establish a performance floor.

<img width="835" height="380" alt="Image" src="https://github.com/user-attachments/assets/748dbc3b-cba3-4c7d-9fb4-19c81f43c703" />

<img width="827" height="377" alt="Image" src="https://github.com/user-attachments/assets/43f38400-d875-4737-a49e-a40d33a1e35f" />


**B. Quantum Kernel SVM (QSVM)** 
Our quantum approach uses QSVMs, computing data similarity via state fidelity using the quantum kernel trick instead of variational training.

- Embedding and Scaling is Critical for Quantum Kernels.
A major technical hurdle we overcame was phase wrapping. Quantum rotation gates are periodic (0≡2π), and raw PCA values often exceed this range, causing data to wrap around the Bloch sphere and collapse the data hierarchy. We addressed this by applying strict feature scaling, mapping all principal components to [0,0.9], ensuring that each data point occupies a distinct position in quantum state space.

**Fourier and QAOA Quantum Feature Maps**
Fourier and QAOA feature maps produce smooth decision boundaries by leveraging phase encoding and interference.

<img width="843" height="357" alt="Image" src="https://github.com/user-attachments/assets/0b5f819b-eca2-475f-a564-48685c9ed248" />

<img width="836" height="393" alt="Image" src="https://github.com/user-attachments/assets/11b57cf1-e191-48b3-ac6a-541b033bbbfc" />

Confusion matrices show strong diagonal dominance for both Fourier and QAOA kernels, indicating accurate class separation comparable to the classical RBF baseline. Decision boundary plots further confirm stable and coherent separation, demonstrating that feature scaling effectively mitigates phase-induced noise.

Overall, these results support the effectiveness of the adopted preprocessing and encoding strategies

**Haar-Random Quantum Kernels**
Haar-random quantum kernels increased expressivity but showed reduced generalization as circuit randomness and qubit count increased, highlighting the limitations of unstructured randomness in low-data regimes.
Haar-random kernel performance for 2- and 3-qubit circuits:

<img width="447" height="142" alt="Image" src="https://github.com/user-attachments/assets/5ec4fff9-d411-45d8-ae61-2e136c3be01f" />

<img width="883" height="338" alt="Image" src="https://github.com/user-attachments/assets/276bdede-ea49-4909-9644-5214ac2f3afb" />

This underscores the importance of structure over randomness in quantum kernel design.



**Trainable Quantum Kernels: Data Re-uploading and HEA**
Data re-uploading reduces boundary fragmentation, while the hardware-efficient ansatz (HEA) yields coherent decision boundaries and reliable class separation

<img width="897" height="411" alt="Image" src="https://github.com/user-attachments/assets/584f7960-4849-46d9-8160-c2b9073dd191" />

<img width="907" height="413" alt="Image" src="https://github.com/user-attachments/assets/809793ae-defa-48e7-9134-29021c8defe1" />
This highlights the importance of trainability and hardware-aligned circuit design for achieving stable quantum classification behavior.

---

## Technical Challenges

Quantum models performed poorly on small datasets, showing high variance and weak generalization. Increased qubits, unstructured circuits, and limited PCA dimensionality (due to compute constraints) further impacted stability, making strong classical SVM baselines difficult to surpass.



---

## Current Status

Classical and quantum kernel models have been successfully implemented, establishing strong baselines and meaningful performance insights.

---

## Path Forward

To complete our MVP and expand our findings, the next phase of development will focus on:

- Further Model Refinement
- **Advanced Architectures:** Explore QFT-based and other variational classifiers beyond kernel methods. We aim to determine if these architectures can offer computational speedups or accuracy improvements over the kernel-based approach.
- **Generalization:** Evaluate optimized kernels on diverse datasets to test robustness.
- **Noise & Hardware Effects:** Compare performance on noiseless and noisy simulators, and where feasible, real hardware.

**Final MVP (End of January):** A complete, reproducible benchmarking framework comparing classical SVMs and multiple quantum models, including kernel-based and QFT-inspired architectures, evaluated across datasets, accompanied by a high-quality blog post or paper summarizing key empirical findings and comparative results.
